logstash-core/src/test/java/org/logstash/config/ir/CompiledPipelineTest.java

```diff
@@ -159,12 +164,56 @@ public void buildsForkedPipeline() throws Exception {
                 filters,
                 Collections.singletonMap("mockoutput", mockOutputSupplier())
             )
-        ).buildExecution().compute(RubyUtil.RUBY.newArray(testEvent), false, false);
+        ).buildExecution(false).compute(RubyUtil.RUBY.newArray(testEvent), false, false);
         final Collection<JrubyEventExtLibrary.RubyEvent> outputEvents = EVENT_SINKS.get(runId);
         MatcherAssert.assertThat(outputEvents.size(), CoreMatchers.is(1));
         MatcherAssert.assertThat(outputEvents.contains(testEvent), CoreMatchers.is(true));
     }
 
+    @Test
+    public void preservesInputEventInstantiationOrder() throws Exception {
+        testEventOrder(true, new int[]{0, 2, 1});
```

Comment:
nitpick: the 2nd argument here is dependent on the implementation of the test in a non-obvious way (e.g., the test constructs the batch in a specific out-of-order manner, then uses the value of this arguement to map output ordering to a desired ordering).

Reply:
I'm open to other ways of implementing the test which includes a comment about why the batch is constructed out-of-order.